rootfs/ingest/python/qserv/metadata.py

```diff
@@ -42,24 +44,137 @@
+    and for a given path
+
+     Store informations which will allow to retrieve contributions file
+     each entry of the list is a tuple: (<path>, [chunk_ids], <is_overlap>, <table>)
+     where [chunk_ids] is the list of the contribution files (XOR overlap) available
+     at a given path for a given table and a given chunk
+    """
+
+    base_path: str
+    """ Base path """
+
+    table: str
+    """ Table name """
+
+    files: List[str]
+    """ Files for regular tables, empty for partitioned tables """
+
+    chunks: List[int]
+    """ Chunks ids for files for partitioned tables, empty for non-partitioned tables """
+
+    chunks_overlap: List[int]
+    """ Chunks ids for overlap files for partioned tables, empty for non-partitioned tables """
+
+    def get_contrib(self):
+        """Generator for contribution specifications for a given table and a given path
+
+        Yields
+        ------
+            Iterator[List[dict()]]: Iterator on each contribution specifications for a table
+        """
+        for file in self.files:
+            data = {
+                "chunk_id": None,
+                "filepath": self._filepath(file),
+                "is_overlap": None,
+                "table": self.table,
+            }
+            yield data
+
+        for id in self.chunks:
+            data = {
+                "chunk_id": id,
+                "filepath": self._filepath(f"chunk_{id}.txt"),
+                "is_overlap": False,
+                "table": self.table,
+            }
+            yield data
+
+        for id in self.chunks_overlap:
+            data = {
+                "chunk_id": id,
+                "filepath": self._filepath(f"chunk_{id}_overlap.txt"),
+                "is_overlap": True,
+                "table": self.table,
+            }
+            yield data
+
+    def _filepath(self, filename: str) -> str:
+        filepath = (self.base_path.strip("/")
+                    + "/"
+                    + filename.strip("/"))
+        return filepath
 
 
-class ContributionMetadata():
+class TableSpec:
+    """Contain table specifications for a given database"""
+
+    contrib_specs: List[TableContributionsSpec]
+    data: List[Any]
+    schema_file: str
+    is_director: bool
+    is_partitioned: bool
+    json_indexes: List[str]
+    json_schema: Dict
+    name: str
+
+    def __init__(self, metadata_url: str, table_meta: Dict):
+        self.data = table_meta["data"]
+        schema_file = table_meta["schema"]
+        self.json_schema = json_get(metadata_url, schema_file)
+        self.name = self.json_schema["table"]
+        self.is_partitioned = self.json_schema["is_partitioned"] == 1
+        self.is_director = (
+            "director_table" in self.json_schema
+            and len(self.json_schema["director_table"]) == 0
+        )
+        idx_files = table_meta["indexes"]
+        self.json_indexes = []
+        for f in idx_files:
+            self.json_indexes.append(json_get(metadata_url, f))
```

Comment:
One-line:

    self.json_indexes = [json_get(metadata_url, f) for f in idx_files]